pkgs/api/src/event_broadcaster.zig

```diff
@@ -0,0 +1,241 @@
+const std = @import("std");
+const Thread = std.Thread;
+const Mutex = Thread.Mutex;
+const events = @import("./events.zig");
+
+/// SSE connection wrapper
+pub const SSEConnection = struct {
+    stream: std.net.Stream,
+    allocator: std.mem.Allocator,
+    mutex: Mutex,
+
+    pub fn init(stream: std.net.Stream, allocator: std.mem.Allocator) SSEConnection {
+        return SSEConnection{
+            .stream = stream,
+            .allocator = allocator,
+            .mutex = Mutex{},
+        };
+    }
+
+    pub fn sendEvent(self: *SSEConnection, event_json: []const u8) !void {
+        self.mutex.lock();
+        defer self.mutex.unlock();
+
+        try self.stream.writeAll(event_json);
+    }
+
+    pub fn deinit(self: *SSEConnection) void {
+        self.stream.close();
+    }
+};
+
+/// Thread-safe event broadcaster for SSE connections
+pub const EventBroadcaster = struct {
+    connections: std.ArrayList(*SSEConnection),
```

Comment:
Why `EventBroadcaster` need manage connections? It shouldn't be its responsibility